crates/bevy_ecs/src/reflect/component.rs

```diff
@@ -180,6 +188,10 @@ impl ReflectComponent {
     }
 
     /// Gets the value of this [`Component`] type from the entity as a mutable reflected reference.
+    ///
+    /// # Panics
```

Comment:
Follow-up: I would prefer to make these all return Results instead of panicking.